simphony/io/tests/test_h5_cuds.py

```diff
@@ -255,5 +259,85 @@ def tearDown(self):
             engine.close()
 
 
+class TestMaterialsOperations(unittest.TestCase):
+
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+        self.filename = os.path.join(self.temp_dir, 'test.cuds')
+        self.addTypeEqualityFunc(
+            Material, partial(compare_materials, testcase=self))
+
+        self.expected = Materials()
+        for i in xrange(5):
+            self.expected.add_material(
+                Material(description="Material {}".format(i),
+                         data=create_data_container()))
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+
+    def file_factory(self):
+        return H5CUDS.open(self.filename)
+
+    def cleanup(self):
+        os.remove(self.filename)
+
+    def test_add_get_materials(self):
+        handle = self.file_factory()
+        expected_materials_dict = {}
+        for material in self.expected.iter_materials():
+            expected_materials_dict[material.uid] = material
+
+        handle.add_materials(self.expected)
+
+        # Closing and reopening the file
+        handle.close()
+        handle = self.file_factory()
+
+        stored = handle.get_materials()
+
+        stored_materials = [material for material in stored.iter_materials()]
+
+        self.assertEqual(len(expected_materials_dict), len(stored_materials))
+        for material in stored_materials:
+            self.assertEqual(
+                material, expected_materials_dict[material.uid])
+
+        handle.close()
+        self.cleanup()
+
+    def test_remove_materials(self):
+        handle = self.file_factory()
+        handle.add_materials(self.expected)
+
+        handle.remove_materials()
+
+        # Closing and reopening the file
+        handle.close()
+        handle = self.file_factory()
+
+        stored = handle.get_materials()
+        stored_materials = [material for material in stored.iter_materials()]
+
+        # The file should not contain any materials
+        self.assertEqual(len(stored_materials), 0)
+
+        handle.close()
+        self.cleanup()
+
+    def test_error_when_adding_existing_materials(self):
+        handle = self.file_factory()
+        handle.add_materials(self.expected)
+
+        handle.close()
+        handle = self.file_factory()
+
+        with self.assertRaises(ValueError):
+            handle.add_materials(self.expected)
+
+        handle.close()
+        self.cleanup()
+
+
```

Comment:
>  In future, Materials will be contained in the SD component, and we'll probably then have add_state_data method in H5CUDS. I understand that in future it should be possible to save many "state datas" in a file, but only one Materials container per state data. This will be a big change to the file layout, and it will probably make add_materials redundant.

Yeah, I see your point.  We are waiting for the `SD` or some sort of `Cuds Model` (name?) but need to be able to save the materials now.  

What about removing `H5Materials` class (but reusing its code) and just have the following methods to get/set a single `Materials`:
```
class H5Cuds ...

def get_materials()  
    ....  # returns Materials
def set_materials(materials) 
    ...  # sets Materials (replaces materials)  
```
(or alternatively don't deal with 'Materials' and just have methods to get/delete/update/add 'Material')